### PR TITLE
Gorina/3192 role criticality and description display in role request dashboard width

### PR DESCRIPTION
**Pull request by EvgeniyaGorina**
https://redmine.czechidm.com/issues/3192

### CODE REVIEW

File: Realization/frontend/czechidm-core/src/content/identity/IdentityRoleTable.js [hunk 1]
```diff
@@ -201,7 +200,7 @@ export class IdentityRoleTable extends Advanced.AbstractTableContent {
         <Advanced.Table
           ref="table"
           uiKey={ this.getUiKey() }
-          columns={ _columns }
+          columns={_columns}
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/identity/IdentityRoleTable.js:203]: The spaces should be there.

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js [hunk 2]
```diff
@@ -721,6 +719,18 @@ export class RequestIdentityRoleTable extends Advanced.AbstractTableContent {
...
+              header={this.i18n('entity.Role.description')}
+              face="text"
+              className='descriptionColumn'
+              cell={ ({ rowIndex, data }) => data[rowIndex]._embedded.role.description }
+            />
+              <Advanced.Column
+            property="priority"
+            header={this.i18n('entity.Role.priority')}
+            face="text"
+            cell={ ({ rowIndex, data }) => data[rowIndex]._embedded.role.priority }/>
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:733]: Please, format this the same way other columns are.

File: Realization/frontend/czechidm-core/src/content/role/RoleFormAttributes.js [hunk 3]
```diff
@@ -131,7 +131,7 @@ class RoleFormAttributes extends Basic.AbstractContent {
           </div>
         </Basic.AbstractForm>
         <Basic.Div rendered={entity.identityRoleAttributeDefinition}>
-          <Basic.ContentHeader icon="fa:th-list" text={ this.i18n('content.role.formAttributes.header') } style={{ marginBottom: 0 }}/>
+          <Basic.ContentHeader icon="fa:th-list" text={ this.i18n('content.role.formAttributes.header') } style={{marginBottom: 0 }}/>
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/role/RoleFormAttributes.js:134]: The space should be there.

File: Realization/frontend/czechidm-core/src/locales/cs.json [hunk 4]
```diff
@@ -742,12 +742,7 @@
         "result": {
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/locales/cs.json:742]: This file shouldn't be changed at all.

File: Realization/frontend/czechidm-core/src/locales/en.json [hunk 5]
```diff
@@ -709,12 +709,7 @@
         "result": {
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/locales/en.json:709]: This file shouldn't be changed at all.

File: Realization/frontend/czechidm-core/src/css/main.less [hunk 6]
```diff
@@ -32,4 +32,27 @@ div.customTextAreNot.form-group {
...
+
+    .spanRole{
+        white-space:normal;
+        display:inline-block;
+     }
+     
+    span > .entity-info{
+        color:#f50057;
+    }
+  }
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/css/main.less:57]: The code formatting is wrong in ".advanced-tree".

In my opinion, this should not be placed in main.less but in a more specific less file (Table.less seems ideal).

**Review by tomdoischer — CHANGES_REQUESTED** [review]
Just minor notes.

File: Realization/frontend/czechidm-core/src/content/identity/IdentityRoleTable.js (hunk 1 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/content/identity/IdentityRoleTable.js:203]: done

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js (hunk 2 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:733]: done

File: Realization/frontend/czechidm-core/src/content/role/RoleFormAttributes.js (hunk 3 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/content/role/RoleFormAttributes.js:134]: done

File: Realization/frontend/czechidm-core/src/locales/cs.json (hunk 4 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/locales/cs.json:742]: done

File: Realization/frontend/czechidm-core/src/locales/en.json (hunk 5 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/locales/en.json:709]: done

File: Realization/frontend/czechidm-core/src/css/main.less (hunk 6 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/css/main.less:57]: Doesn't work from other less, I have try it but visible for components those styles only from main, have no idea why

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js [hunk 7]
```diff
@@ -860,4 +881,31 @@ RequestIdentityRoleTable.defaultProps = {
...
+    'automaticRole',
+    'candicateUsers',
+    'action'
+  ])
+};
+function select(state, component) {
+  return {
+    columns: component.columns || ConfigurationManager.getPublicValueAsArray(
+      state,
+      'idm.pub.app.show.RequestIdentityRoleTable.table.columns',
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:906]: I suggest the configuration key be: idm.pub.app.show.role.request.table.columns

File: Realization/frontend/czechidm-core/src/css/RoleTable.less [hunk 8]
```diff
@@ -0,0 +1,24 @@
+
+//Wide of columns in role and description and priority
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/css/RoleTable.less:2]: This is not the right location for the file. This file should be in Realization/frontend/czechidm-core/src/content/role

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js [hunk 9]
```diff
@@ -860,4 +881,31 @@ RequestIdentityRoleTable.defaultProps = {
   showEnvironment: true
 };
 
-export default RequestIdentityRoleTable;
+RequestIdentityRoleTable.defaultProps = {
+  columns: ConfigLoader.getConfig('RequestIdentityRoleTable.table.columns', [
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:885]: This should be 'role.request.table.columns'

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js [hunk 10]
```diff
@@ -860,4 +881,31 @@ RequestIdentityRoleTable.defaultProps = {
...
+    'role.name',
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:886]: This should just be 'name', you will need to change the column use a little bit (basically the same way you did for priority and description).

The same for the examples below:
'_embedded.role.baseCode', -> baseCode
'_embedded.role.environment', -> environment

**Review by tomdoischer — CHANGES_REQUESTED** [review]
I have minor notes. Also, please edit the following file

/idm-app/src/main/resources/application.properties

and add the default configuration of columns for the role request table in the same way that 
"idm.pub.app.show.role.table.columns=name, baseCode, environment, disabled, description" is used.

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js (hunk 7 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:906]: done

File: Realization/frontend/czechidm-core/src/css/RoleTable.less (hunk 8 above)
Comment (EvgeniyaGorina) [reply Realization/frontend/czechidm-core/src/css/RoleTable.less:2]: done

File: Realization/backend/app/src/main/resources/application.properties [hunk 11]
```diff
@@ -376,6 +376,23 @@ idm.pub.app.show.role.baseCode=true
...
+# - validFrom
+# - validTill
+# - directRole
+# - automaticRole
+# - baseCode
+# - systemState
+# - environment
+# - priority
+# - candicateUsers
+# - action
```
Comment (tomdoischer) [inline Realization/backend/app/src/main/resources/application.properties:394]: The default is great, just please add "action" to the end. It is now missing and users cannot remove roles.

File: Realization/backend/app/src/main/resources/application.properties [hunk 12]
```diff
@@ -376,6 +376,23 @@ idm.pub.app.show.role.baseCode=true
 # - disabled
 # - description
 idm.pub.app.show.role.table.columns=name, baseCode, environment, disabled, description
+# Rendered column in role table agenda. Comma is used as separator. Order of rendered columns is preserved as configured.
```
Comment (tomdoischer) [inline Realization/backend/app/src/main/resources/application.properties:379]: "Rendered column in role table agenda." should be "Rendered columns in role request table agenda."

**Review by tomdoischer — APPROVED** [review]
Looks good to me!

**Pull request merged**